### PR TITLE
Batched training and silent

### PR DESCRIPTION
**Pull request by vossenv**
Add batch process functionality:

- simple approach to looping the train function over a number of files
- output audio field type updated to allow multiple file selection or single file (only a single input source can be used)
- 'modelname' added in some places to propagate the name of the final ".nam" file, so that multiple files can be dumped without overwriting each other.  Modelname is just the filename.

Add two new supporting options to GUI interface:

- automatically save plots to png next to model
- 'silent' run mode which will not prompt the user to close the delay window or final plot before finishing

multi1

https://user-images.githubusercontent.com/44332958/226157827-418f7ca3-f68d-4e10-a1c4-82323f168c4b.mp4

### CODE REVIEW

**Comment by 2-dor** [comment]
> Add batch process functionality:
> 
> * simple approach to looping the train function over a number of files
> * output audio field type updated to allow multiple file selection or single file (only a single input source can be used)
> * 'modelname' added in some places to propagate the name of the final ".nam" file, so that multiple files can be dumped without overwriting each other.  Modelname is just the filename.
> 
> Add two new supporting options to GUI interface:
> 
> * automatically save plots to png next to model
> * 'silent' run mode which will not prompt the user to close the delay window or final plot before finishing
> 
> multi1

This is super cool!

**Comment by vossenv** [comment]
> > Add batch process functionality:
> > 
> > * simple approach to looping the train function over a number of files
> > * output audio field type updated to allow multiple file selection or single file (only a single input source can be used)
> > * 'modelname' added in some places to propagate the name of the final ".nam" file, so that multiple files can be dumped without overwriting each other.  Modelname is just the filename.
> > 
> > Add two new supporting options to GUI interface:
> > 
> > * automatically save plots to png next to model
> > * 'silent' run mode which will not prompt the user to close the delay window or final plot before finishing
> > 
> > multi1
> 
> This is super cool!

Thanks :) this is @northern_fox btw

**Comment by 2-dor** [comment]
> > > Add batch process functionality:
> > > 
> > > * simple approach to looping the train function over a number of files
> > > * output audio field type updated to allow multiple file selection or single file (only a single input source can be used)
> > > * 'modelname' added in some places to propagate the name of the final ".nam" file, so that multiple files can be dumped without overwriting each other.  Modelname is just the filename.
> > > 
> > > Add two new supporting options to GUI interface:
> > > 
> > > * automatically save plots to png next to model
> > > * 'silent' run mode which will not prompt the user to close the delay window or final plot before finishing
> > > 
> > > multi1
> > 
> > 
> > This is super cool!
> 
> Thanks :) this is @northern_fox btw

Yeah - I know! Great stuff!

File: nam/train/gui.py [hunk 1]
```diff
@@ -191,6 +198,30 @@ def __init__(self):
 
         self._check_button_states()
 
+    def getAdditionalOptionsFrames(self):
```
Comment (sdatkinson) [inline nam/train/gui.py:201]: Nit: can we make this function name PEPpy & private in scope? (`_get_additional_options_frames` plz)

File: nam/train/gui.py [hunk 2]
```diff
@@ -112,6 +115,7 @@ def _set_val(self):
         res = {
             _PathType.FILE: filedialog.askopenfilename,
             _PathType.DIRECTORY: filedialog.askdirectory,
+            _PathType.MULTIFILE: filedialog.askopenfilenames,
```
Comment (sdatkinson) [inline nam/train/gui.py:118]: Nit: I'd love if the tuple were stripped in the displayed text if only one output file were selected.

**Review by sdatkinson — APPROVED** [review]
Awesome! Love the video demo too--much appreciated ❤️ 

A few nits; lmk if you can get them.

If not, I'd still be fine to approve & merge, so marking as approved.

File: nam/train/core.py [hunk 3]
```diff
@@ -308,7 +310,12 @@ def _esr(pred: torch.Tensor, target: torch.Tensor) -> float:
 
 
 def _plot(
-    model, ds, window_start: Optional[int] = None, window_end: Optional[int] = None
+        model,
```
[the remark this answers is not in the document]
Comment (vossenv) [reply nam/train/core.py:313]: Neat! Did not know about that

File: nam/train/gui.py (hunk 2 above)
Comment (vossenv) [reply nam/train/gui.py:120]: Done - still displays tuple format for multiple files

File: nam/train/gui.py [hunk 4]
```diff
@@ -38,7 +41,6 @@ def _ensure_graceful_shutdowns():
     _install_is_valid = True
 except ImportError:
     _install_is_valid = False
-
```
Comment (vossenv) [inline nam/train/gui.py:41]: This change added by black

**Comment by vossenv** [comment]
Multiple files

image

Single file

image

I haven't tried this on macOS, btw

**Review by sdatkinson — APPROVED** [review]
LGTM! 👍🏻 

Thanks!

**Pull request merged**